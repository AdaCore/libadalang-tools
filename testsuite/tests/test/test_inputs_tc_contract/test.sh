#!/bin/bash

echo "generate test driver"
gnattest -q -P contracts.gpr
echo "generate test driver with instrumentation"
gnattest -q -P contracts.gpr --dump-test-inputs
echo "build instrumented test driver"
gprbuild -q -P gnattest/harness/test_driver.gpr --src-subdirs=gnattest-instr --implicit-with=gnattest/harness/tgen_support/tgen_support.gpr

gnattest/harness/test_runner
if ls ./sqrt-*-2 1> /dev/null 2>&1; then
    echo "2nd test input present"
fi
