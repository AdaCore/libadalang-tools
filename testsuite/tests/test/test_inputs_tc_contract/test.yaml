description:
    Test that test routines for different Test_Case
    of the same subprogram do not overwrite each other.

driver: gnattest_driver

