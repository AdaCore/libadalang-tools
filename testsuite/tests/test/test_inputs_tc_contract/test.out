generate test driver
generate test driver with instrumentation
build instrumented test driver
2 tests run: 2 passed; 0 failed; 0 crashed.
2nd test input present
